### ISSUE TITLE
A property named "time" raises DriverSupportError for ESRI Shapefiles

### ISSUE THREAD

**Issue by mdklatt** [Toblerity/Fiona#692]
## Expected behavior and actual behavior.

From https://github.com/Toblerity/Fiona/issues/572, I see that properties of *type* `datetime.time` are not supported for ESRI Shapefiles, but surely a `str` property *named* "time" is acceptable?

Fiona raises `DriverSupportError: ESRI Shapefile does not support time fields` for schemas that have a property named "time".

## Steps to reproduce the problem.

Running this code:

``` python
import fiona

schema = {
    "geometry": "Point",
    "properties": {
        "time": "str",  # type is str, should be okay
    }
}

c = fiona.collection("time_bug.shp", "w", "ESRI Shapefile", schema)
```

Gives this result:
``` console
Traceback (most recent call last):
  File "/vagrant/fiona_time_bug.py", line 13, in <module>
    c = fiona.collection("time_bug.shp", "w", "ESRI Shapefile", schema)
  File "/opt/DataPortal/lib/python3.6/site-packages/fiona/env.py", line 405, in wrapper
    return f(*args, **kwargs)
  File "/opt/DataPortal/lib/python3.6/site-packages/fiona/__init__.py", line 263, in open
    **kwargs)
  File "/opt/DataPortal/lib/python3.6/site-packages/fiona/collection.py", line 145, in __init__
    self._check_schema_driver_support()
  File "/opt/DataPortal/lib/python3.6/site-packages/fiona/collection.py", line 419, in _check_schema_driver_support
    raise DriverSupportError("ESRI Shapefile does not support time fields")
fiona.errors.DriverSupportError: ESRI Shapefile does not support time fields

```

## Operating system

Linux-3.10.0-514.el7.x86_64-x86_64-with-centos-7.3.1611-Core
Python 3.6.5

## Fiona and GDAL version and provenance

Installed from PyPI via pip:

Fiona 1.8.3
GDAL 2.3.2

**Comment by mdklatt** [comment]
The source of the exception is [collection.py](https://github.com/Toblerity/Fiona/blob/master/fiona/collection.py#L418):

``` python
def _check_schema_driver_support(self):
        """Check support for the schema against the driver

        See GH#572 for discussion.
        """
        gdal_version_major = get_gdal_version_num() // 1000000
        for field in self._schema["properties"]:
            field_type = field.split(":")[0]
            if self._driver == "ESRI Shapefile":
                if field_type == "datetime":
                    raise DriverSupportError("ESRI Shapefile does not support datetime fields")
                elif field_type == "time":
                    raise DriverSupportError("ESRI Shapefile does not support time fields")
```

It looks like the problem is with`field_type = field.split(":")[0]`. The intent of this seems to be to split the property's type from its width, but `field` is the property *name*, right?

**Comment by sgillies** [comment]
@mdklatt yes, you're right. We'll fix this in the next release. Next week, I think.

**Comment by snorfalorpagus** [comment]
I think this should be:

```
for field in self._schema["properties"].values():
```

**Comment by sgillies** [comment]
Resolved by #695.

**Comment by karthikaditya** [comment]
I'm using the 1.8.13 version of Fiona installed from pip and I get this error when trying to create a shapefile. Can anyone please clarify if this fix is present in the latest release?

**Comment by mdklatt** [comment]
I can confirm that this is fixed for *v1.8.13.post1*.
